resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverBuilder.java

```diff
@@ -33,24 +33,24 @@
 /**
  * A {@link DnsNameResolver} builder.
  */
-public final class DnsNameResolverBuilder {
-
-    private final EventLoop eventLoop;
-    private ChannelFactory<? extends DatagramChannel> channelFactory;
-    private InetSocketAddress localAddress = DnsNameResolver.ANY_LOCAL_ADDR;
-    private DnsServerAddresses nameServerAddresses;
-    private DnsCache resolveCache;
-    private Integer minTtl;
-    private Integer maxTtl;
-    private Integer negativeTtl;
-    private long queryTimeoutMillis = 5000;
-    private InternetProtocolFamily[] resolvedAddressTypes = DnsNameResolver.DEFAULT_RESOLVE_ADDRESS_TYPES;
-    private boolean recursionDesired = true;
-    private int maxQueriesPerResolve = 3;
-    private boolean traceEnabled;
-    private int maxPayloadSize = 4096;
-    private boolean optResourceEnabled = true;
-    private HostsFileEntriesResolver hostsFileEntriesResolver = HostsFileEntriesResolver.DEFAULT;
+public abstract class DnsNameResolverBuilder<T extends DnsNameResolverBuilder<T>> {
+
+    protected final EventLoop eventLoop;
+    protected ChannelFactory<? extends DatagramChannel> channelFactory;
+    protected InetSocketAddress localAddress = DnsNameResolver.ANY_LOCAL_ADDR;
+    protected DnsServerAddresses nameServerAddresses = DnsServerAddresses.defaultAddresses();
+    protected DnsCache resolveCache;
+    protected Integer minTtl;
+    protected Integer maxTtl;
+    protected Integer negativeTtl;
+    protected long queryTimeoutMillis = 5000;
+    protected InternetProtocolFamily[] resolvedAddressTypes = DnsNameResolver.DEFAULT_RESOLVE_ADDRESS_TYPES;
+    protected boolean recursionDesired = true;
+    protected int maxQueriesPerResolve = 3;
+    protected boolean traceEnabled;
+    protected int maxPayloadSize = 4096;
+    protected boolean optResourceEnabled = true;
+    protected HostsFileEntriesResolver hostsFileEntriesResolver = HostsFileEntriesResolver.DEFAULT;
```

Comment:
Could we instead add the protected getters, like we do in `AbstractHttp2ConnectionHandlerBuilder`?